qa/rpc-tests/wallet_mergetoaddress_2.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import shutil
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import assert_equal, initialize_chain_clean, \
+    start_node, connect_nodes_bi, sync_blocks, sync_mempools, \
+    wait_and_assert_operationid_status
+
+import time
+from decimal import Decimal
+from distutils.dir_util import copy_tree
+
+class WalletMergeToAddressTest (BitcoinTestFramework):
+
+    op_result_k = "op_result"
+    balance_delta_k = "balance_delta"
+    joinsplit_len_k = "joinsplit_len"
+    mergingUTXOs_k = "mergingUTXOs"
+    mergingNotes_k = "mergingNotes"
+    mergingTransparentValue_k = "mergingTransparentValue"
+    mergingShieldedValue_k = "mergingShieldedValue"
+
+    def import_data_to_data_dir(self):
+        #importing datadir resource
+        import os
+        os.chdir(os.path.dirname(__file__))
+        resource_file = os.getcwd() + "/resources/wallet_mergetoaddress_2/test_setup_.zip"
+        import zipfile
+        with zipfile.ZipFile(resource_file, 'r') as zip_ref:
+            zip_ref.extractall(self.options.tmpdir)
+        copy_tree(self.options.tmpdir + "/test_setup_", self.options.tmpdir)
+        shutil.rmtree(self.options.tmpdir + "/test_setup_")
+
+    def generate_notes(self, quantity):
+        myzaddr0 = self.nodes[0].z_getnewaddress()
+        results = []
+        for i in range(quantity):
+            self.nodes[0].generate(1)
+            self.sync_all()
+            self.nodes[1].generate(100)
+            self.sync_all()
+            results.append(self.nodes[0].z_shieldcoinbase("*", myzaddr0, 0))
+            wait_and_assert_operationid_status(self.nodes[0], results[i]['opid'])
+        self.nodes[1].generate(1)
+        self.sync_all()
+
+    def send_transparent(self, quantity):
+        for i in range(quantity):
+            self.nodes[1].sendmany("", {self.nodes[0].listaddresses()[0]: 10}, 100)
+        self.nodes[1].generate(100)
+        self.sync_all
+
+    def get_merged_transaction(self, unspent_transactions_before_merge, unspent_transactions_after_merge):
+        txid = unspent_transactions_after_merge[0]["txid"]
+        for aft in range(len(unspent_transactions_after_merge)):
+            for bef in range(len(unspent_transactions_before_merge)):
+                if (unspent_transactions_after_merge[aft]["txid"] == unspent_transactions_before_merge[bef]["txid"]):
+                    break
+                if (bef == len(unspent_transactions_before_merge) - 1):
+                    txid = unspent_transactions_after_merge[aft]["txid"]
+        return self.nodes[2].gettransaction(txid)
```

Comment:
This can be simplified using the `in` keyword to check if a specific element is present in a list or dictionary
```suggestion
    def get_merged_transaction(self, unspent_transactions_before_merge, unspent_transactions_after_merge):
        for tx in unspent_transactions_after_merge:
            if tx not in unspent_transactions_before_merge:
                return self.nodes[2].gettransaction(tx["txid"])
```

Reply:
As a follow up, the previous suggested change compares the txs in both lists. If you need to compare only the "txid" of each transaction, the function should be this one:
```suggestion
    def get_merged_transaction(self, unspent_transactions_before_merge, unspent_transactions_after_merge):
        unspent_txid_before = [x["txid"] for x in unspent_transactions_before_merge]
        for txid in [x["txid"] for x in unspent_transactions_after_merge]:
            if txid not in unspent_txid_before:
                return self.nodes[2].gettransaction(txid)
        return None
```

Reply:
Done